### ISSUE TITLE
Choose whether screenshot should include HTML page or not on an on-demand basis

### ISSUE THREAD

**Issue by kenston** [selenide/selenide#436]
The option `com.codeborne.selenide.Configuration.savePageSource` allows us to decide whether HTML source file should be saved or not. This seems to be a one-time decision we have to make during runtime.

I'm planning to save disk usage by setting it to `false` by default; however, I may need to save the HTML source file during failed interactions. This means I have to set `savePageSource` back to `true`, take a screenshot, then set `savePageSource` back to `false` while considering thread synchronization/concurrency.

Is it possible for us to expose an option parameter instead?

**Comment by asolntsev** [comment]
@kenston At a first glance, it seems to be a huge overengineering to me.
1. Do you really need to save disk space so much? How many kilobytes will you save? Does this space matter? Isn't it simple just to archive old builds (you know, Jenkins has an option to keep only last N builds)?
2. If you know exactly what tests are expected to fail, why not just fix them?

**Comment by BorisOsipov** [comment]
@kenston maybe ability to have public method `File savePageSourceToFile(String fileName, WebDriver webdriver)`  is good enough for your case? of cource method should save page source without paying attention to the option `com.codeborne.selenide.Configuration.savePageSource`

**Comment by kenston** [comment]
@asolntsev (1) the HTML pages are bigger than the image screenshots and these pages are only checked when there's an issue/error. We sometimes download the entire zip file and it is just unnecessary to have them. (2) To fix them, we may sometimes need to look at the HTML source file. We only want to keep the source file during an interaction error.

@BorisOsipov that looks good. We'll turn off `savePageSource` and use the `savePageSourceToFile` method as needed

**Comment by asolntsev** [comment]
@kenston I see that we have such a method:

```java
String fileName = Selenide.screenshot("the-name-of-file-here");
```

This method creates two files:
* `the-name-of-file-here.png`
* `the-name-of-file-here.html`

Isn't it what you needed?

**Comment by kenston** [comment]
@asolntsev, I wonder if it's possible to create only `the-name-of-file-here.png` but not `the-name-of-file-here.html` on certain/specific cases. In my first post, I highlighted that the setting is currently a global one.

**Comment by asolntsev** [comment]
@kenston Wait, I've messed up. If you want to create only `the-name-of-file-here.png` but not `the-name-of-file-here.html` on certain/specific cases, you just need to set `Configuration.savePageSource=false`. Nothing else needed.

**Comment by kenston** [comment]
@asolntsev, yes, but that option is a global one, as I've mentioned in my first port:

> The option com.codeborne.selenide.Configuration.savePageSource allows us to decide whether HTML source file should be saved or not. This seems to be a one-time decision we have to make during runtime.

I was wondering that time if it's possible to decide in a case to case basis.

**Comment by asolntsev** [comment]
@kenston Sorry, I still don't get your idea.
1. If html files take too much disk space, just disable them globally by `Configuration.savePageSource = false`
2. If you still wish to save html files by default, and only disable them in certain tests, then it will not give you any benefit, because number of such tests cannot be big, right? Otherwise you would need to manually add command for creating html files to many tests - it would be too big manual work.

**Comment by kenston** [comment]
@asolntsev, at the time when I opened the ticket, I actually have this in mind:
1. Disable saving HTMLs by default.
2. When taking screenshot, it will only take the image of the page and not the source.
3. In exceptional cases (e.g., during errors), take the image of the page along with the HTML source file.

It doesn't take too much work since I took screenshots before transitioning from one page to the other, but whether that screenshot includes HTML source or not depends on whether an interaction error has occurred or not.

But we can also close the ticket, as the need back then isn't no longer true for me today 😉, but I was explaining the need during that time. Thanks!
